src/config.c

```diff
@@ -119,34 +124,127 @@ void fastd_config_mac(const char *name, const char *impl) {
 			impl, name, name);
 }
 
-/** Handles the configuration of a bind address */
-void fastd_config_bind_address(const fastd_peer_address_t *address, const char *bindtodev, unsigned flags) {
-#ifndef USE_BINDTODEVICE
-	if (bindtodev && !fastd_peer_address_is_v6_ll(address))
-		exit_error("config error: device bind configuration not supported on this system");
+/** Normalization of a bind address, initializing the target address structure */
+static void normalize_address(fastd_peer_address_t *address, const fastd_peer_address_t *config, const char *bindtodev) {
```

Comment:
The `normalize_address()` approach is problematic. Interface names must not be resolved to indices during configuration, as index changes should not require a restart of fastd.